string parent_contact
string obstacle_contact

geometry_msgs/Point[] position
geometry_msgs/Point[] normal
float64[] depth